sympy/utilities/lambdify.py

```diff
@@ -737,7 +737,7 @@ def _lambdifygenerated(x):
     from sympy.core.symbol import Symbol
 
     # If the user hasn't specified any modules, use what is available.
-    if modules is None:
+    if modules is None or hasattr(modules, '__len__') and len(modules) == 0:
```

Comment:
```suggestion
    if modules:
```
The above will be false if the `modules` is an empty container or `None`.

Reply:
Or where possible use anything in the check except `hasattr` and stuff.

Reply:
Then it should be `if not modules`, right?

Reply:
Oh yeah, right. I mixed it up. Though is there anything in the documentation which specifies what all types `modules` can take.